libraries/AP_Scheduler/PerfInfo.cpp

```diff
@@ -29,6 +29,35 @@ void AP::PerfInfo::ignore_this_loop()
     ignore_loop = true;
 }
 
+// allocate the task info array
+void AP::PerfInfo::allocate_task_info(uint8_t num_tasks)
+{
+    task_info = new TaskInfo[num_tasks];
+    if (task_info == nullptr) {
+        AP_HAL::panic("Unable to allocate scheduler TaskInfo");
```

Comment:
I don't think we should panic. It would make the board hard to recover. Just a console printf I think

Reply:
Done!